### ISSUE TITLE
LXD issue when build multiple charms in parallel

### ISSUE THREAD

**Issue by beliaev-maksim** [canonical/craft-providers#391]
### Bug Description

issue in LXD when build multiple projects in parallel

### To Reproduce

https://github.com/canonical/knative-operators/actions/runs/6024286376/job/16429412492

### Relevant log output

```shell
charmcraft internal error: LXDError(brief='Failed to wait for snap refreshes to complete.\n* Command that failed: "lxc --project charmcraft exec local:charmcraft-knative-operator-146046-0-0-amd64 -- env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin CHARMCRAFT_MANAGED_MODE=1 snap watch \'--last=auto-refresh?\'"\n* Command exit code: 1\n* Command standard error output: b\'error: daemon is stopping to wait for socket activation\\n\'', details=None, resolution=None)
```

**Comment by sergiusens** [comment]
Duplicate of #341

**Comment by beliaev-maksim** [comment]
@sergiusens are you sure?

I tested with the edge and see that error after #341 got merged

**Comment by beliaev-maksim** [comment]
rerunning to validate on edge https://github.com/canonical/knative-operators/actions/runs/6024286376?pr=95

**Comment by beliaev-maksim** [comment]
looks to be fixed on edge

**Comment by lengau** [comment]
Reopening this as @PietroPasotti is seeing this in modern Charmcraft

**Comment by PietroPasotti** [comment]
seen on the CI in [this branch](https://github.com/canonical/parca-agent-operator/pull/55)
